randovania/gui/widgets/game_connection_window.py

```diff
@@ -309,11 +311,14 @@ async def _attempt_join(self, builder: ConnectorBuilder):
             return
 
         session_id = self._check_session_data(layout_uuid)
-        try:
-            await self.network_client.listen_to_session(session_id, True)
-            await self.window_manager.ensure_multiplayer_session_window(self.network_client, session_id, self.options)
-        except error.NotAuthorizedForActionError:
-            await async_dialog.warning(self, "Unauthorized", "You're not a member of this session.")
+        if session_id:
```

Comment:
I feel that not having a session id for a session you're trying to login is a considerable severe error that shouldn't be just ignored. It's kinda of "the button should have been disabled instead" kind of situation, so maybe this should just assert.